unmarshal.go

```diff
@@ -191,6 +191,15 @@ func (ro *resourceObject) unmarshalFields(v any, m *Unmarshaler) error {
 		if err != nil {
 			return err
 		}
+		// If there is an embedded struct we want to process the fields in that struct recursively
+		// only if there isn't a relationship defined on the field.
+		if fv.Kind() == reflect.Struct && jsonapiTag == nil {
```

Comment:
Overall the PR looks fine but for this condition, can we change it to the following, and verify it still works?:

```go
if jsonapiTag == nil {
	if ft.Anonymous && derefValue(fv).Kind() == reflect.Struct {
		if err := ro.unmarshalFields( ... ); err != nil {
			return err
		}
	}
	continue
}
```